website/files/models/osfstorage.py

```diff
@@ -307,6 +312,14 @@ def is_checked_out(self):
                 return True
         return False
 
+    @property
+    def is_preprint_primary(self):
```

Comment:
This should be more like "contains_undeletable_item()" and should check for any reason that it's not deletable.